spec/pg/encoder_spec.cr

```diff
@@ -32,18 +39,23 @@ describe PG::Driver, "encoder" do
   test_insert_and_read "int4", EncoderSpec::Status::Open
   test_insert_and_read "int4", EncoderSpec::Status::Closed
 
+  test_insert_and_read "bool[]", [] of Bool
+  test_insert_and_read "bool[]", [true]
+  test_insert_and_read "bool[]", [false]
   test_insert_and_read "bool[]", [true, false, true]
 
   test_insert_and_read "float[]", [1.2, 3.4, 5.6]
 
   test_insert_and_read "integer[]", [] of Int32
   test_insert_and_read "integer[]", [1, 2, 3]
-  test_insert_and_read "integer[]", [[1, 2], [3, 4]]
+  test_insert_and_read "integer[][]", [[1, 2], [3, 4]]
+  test_insert_and_read "integer[][][]", [[[1, 2], [3, 4]], [[5, 6], [7, 8]]]
```

Comment:
I was actually surprised that the `integer[]` example worked before with a nested array. It still does when encoded as binary, too, but I figured I would make this more explicit.